### PR TITLE
Add Non-CanvasSlate.shader

### PR DESCRIPTION
**Pull request by sgreenlay-msft**
## Overview
Adds Non-CanvasSlate.shader which takes an external texture and adds rounded corners and optional gaze/touch cursors.

### CODE REVIEW

**Comment by Zee2** [comment]
We'll definitely want a Canvas variant of this one.

File: com.microsoft.mrtk.graphicstools.unity/Runtime/Shaders/Non-CanvasSlate.shader [hunk 1]
```diff
@@ -1,7 +1,7 @@
 // Copyright (c) Microsoft Corporation.
 // Licensed under the MIT License.
 
-Shader "Non-CanvasSlate" {
+Shader "Graphics Tools/Non-Canvas/Slate" {
```
Comment (sgreenlay-msft) [inline com.microsoft.mrtk.graphicstools.unity/Runtime/Shaders/Non-CanvasSlate.shader:4]: Any preference for Slate vs SlateContent for the name?

Comment (Cameron-Micka) [reply com.microsoft.mrtk.graphicstools.unity/Runtime/Shaders/Non-CanvasSlate.shader:4]: Slate is cool by me if this is _the_ slate shader. We have a [slate proximity ](https://github.com/microsoft/MixedReality-GraphicsTools-Unity/blob/main/com.microsoft.mrtk.graphicstools.unity/Runtime/Shaders/Non-CanvasSlateProximity.shader)shader, but that is just for drawing proximity shadows.

**Review by Cameron-Micka — APPROVED** [review]
Tested locally and looks great: 
image

In a follow up PR we'll add samples/docs/etc.

**Review by joiverso — COMMENTED** [review]
- It would be more efficient to implement the cursors using the same technique as the slate proximity effect so the cost of rendering those is only incurred when needed instead of every pixel covered by every slate.  I have a NonCanvasTexturedCursor shader that might be easily adapted for that.
- The base shader is out-of-date w.r.t. to the backplate shaders (no color gradient)
- We could easily add a texture option to the existing backplate shader.
I think it'd be better to re-use the existing backplate shaders along with cursor affordances when required instead of maintaining an extra set of shaders just for slates.

**Comment by Cameron-Micka** [comment]
> * It would be more efficient to implement the cursors using the same technique as the slate proximity effect so the cost of rendering those is only incurred when needed instead of every pixel covered by every slate.  I have a NonCanvasTexturedCursor shader that might be easily adapted for that.
> * The base shader is out-of-date w.r.t. to the backplate shaders (no color gradient)
> * We could easily add a texture option to the existing backplate shader.
>   I think it'd be better to re-use the existing backplate shaders along with cursor affordances when required instead of maintaining an extra set of shaders just for slates.

These ideas all sound great to me. @joiverso can you work with @sgreenlay-msft to figure out how to best support the features his team needs?

**Comment by Cameron-Micka** [comment]
@sgreenlay-msft and @joiverso would you like me to close this PR for now while you work on the above bullet points?

**Pull request closed without merging**